opencv=webos4 qtbase=webos101

sparkleholic opened the pull request:
:Release Notes:
Support OpenCV build with Qt6

:Detailed Notes:
opencv=webos4

Although opencv cmake has an option to build with
Qt, the opencv recipe in meta-oe layer didn't have a proper package configuration.
And while enabling WITH_QT for opencv build, the
 following subsequent changes were required.
Here is a cmake snippet of 'opencv/cmake/OpenCVFindLibsGUI.cmake'.
```
macro(ocv_find_package_Qt OCV_QT_VER)
  find_package(Qt${OCV_QT_VER} COMPONENTS Core Gui Widgets Test Concurrent ${ARGN} NO_MODULE)
```
Because opencv needs a Widget module, changed the qtbase recipe (qtbase=webos101).

Second is opencv's contrib extra module's one.
Here is a cmake snippet of 'contrib/modules/cvv/CMakeLists.txt'.
```
if(QT_VERSION_MAJOR EQUAL 6)
  list(APPEND CVV_QT_MODULES Core5Compat)
endif()
```
opencv's contrib module needs a Core5Compat.

Third, the previously fetched 'contrib' was 4.5.5 tag which was not built with Qt6. A patch to support Qt6 was merged in contrib upstream recently, but it was not tagged yet. So I changed the 'contrib' source to the code commit hash with that patch applied to support building with Qt6.

:Testing Performed:
- Locally tested on virtualbox emulator.

:QA Notes:
N/A

:Issues Addressed:
N/A

Change-Id: Ia6c60527c9a778b6a4ded2880e7d049b2d4d3af3

sparkleholic committed:
opencv=webos4 qtbase=webos101

:Release Notes:
Support OpenCV build with Qt6

:Detailed Notes:
opencv=webos4

Although opencv cmake has an option to build with
Qt, the opencv recipe in meta-oe layer didn't have
a proper package configuration.
And while enabling WITH_QT for opencv build, the
 following subsequent changes were required.
Here is a cmake snippet of 'opencv/cmake/OpenCVFindLibsGUI.cmake'.
```
macro(ocv_find_package_Qt OCV_QT_VER)
  find_package(Qt${OCV_QT_VER} COMPONENTS Core Gui Widgets Test Concurrent ${ARGN} NO_MODULE)
```
Because opencv needs a Widget module, changed the qtbase recipe (qtbase=webos101).

Second is opencv's contrib extra module's one.
Here is a cmake snippet of 'contrib/modules/cvv/CMakeLists.txt'.
```
if(QT_VERSION_MAJOR EQUAL 6)
  list(APPEND CVV_QT_MODULES Core5Compat)
endif()
```
opencv's contrib module needs a Core5Compat.

Third, the previously fetched 'contrib' was 4.5.5 tag which was not built with Qt6.
A patch to support Qt6 was merged in contrib upstream recently, but it was not tagged yet.
So I changed the 'contrib' source to the code commit hash with that patch applied to support building with Qt6.

:Testing Performed:
- Locally tested on virtualbox emulator.

:QA Notes:
N/A

:Issues Addressed:
N/A

Change-Id: Ia6c60527c9a778b6a4ded2880e7d049b2d4d3af3

sparkleholic commented:
@shr-project
This recipe is for building `opencv` to work with `Qt6`.
Because `opencv with qt` build requires `QtWidgets` module, I reverted the qtbase recipe to use `widgets PACKAGECONFIG`. 
Could you let me know why we're removing the widget module from qt in webOS build?

sparkleholic committed:
inhert qt6_cmake.bbclass instead of manual setting of QT_HOST_PATH

Change-Id: I1c2bb1ec02ef74171d08f5a3d108051f07708def

shr-project commented:
> @shr-project This recipe is for building `opencv` to work with `Qt6`. Because `opencv with qt` build requires `QtWidgets` module, I reverted the qtbase recipe to use `widgets PACKAGECONFIG`. Could you let me know why we're removing the widget module from qt in webOS build?

It was removed by Elvis Lee in 2016 with commit message saying

"Remove usage of QtWidget module to save system memory and build time."

See http://gpro.lge.com/138034 if you have access to that.
[PLAT-18000] Remove build and install for Qt5Widget

sparkleholic commented:
@shr-project 
Thanks a lot! :) I'll check the link you've shared!

sparkleholic committed:
addressed qt5 build compatibility

Change-Id: I24576fbcf5d9a001d9ab0491f8a5a0d35ff20e5f

sparkleholic committed:
added a blank line for readability

Change-Id: I7492a5bb7fb71457b1ce8dce5bf29ca24d9eb366

sparkleholic committed:
qt5compat=webos1

:Release Notes:
Exclude qtgraphicaleffects from qt5compat

:Detailed Notes:
qt5compat=webos1

Qt6 provides qtgraphicaleffects in the qt5compat package instead of a separate package.
webOS uses a separate recipe to include qtgraphical effects.
This cause a problem installing same files in rootfs when both qt5compat and qtgraphicaleffects are used.
Hence this patch solve the issue by excluding qtgraphicaleffects from qt5compat.
Additional review is needed to see if qt5compat operates normally after applying this patch in the future.

:Testing Performed:
N/A

:QA Notes:
N/A

:Issues Addressed:
N/A
